oxd-server/src/main/java/org/gluu/oxd/server/op/OpClientFactoryImpl.java

```diff
@@ -0,0 +1,68 @@
+package org.gluu.oxd.server.op;
+
+import org.gluu.oxauth.client.*;
+import org.gluu.oxauth.client.uma.UmaClientFactory;
+import org.gluu.oxauth.model.jwt.Jwt;
+import org.gluu.oxd.rs.protect.resteasy.PatProvider;
+import org.gluu.oxd.rs.protect.resteasy.ResourceRegistrar;
+import org.gluu.oxd.rs.protect.resteasy.RptPreProcessInterceptor;
+import org.gluu.oxd.rs.protect.resteasy.ServiceProvider;
+import org.gluu.oxd.server.introspection.ClientFactory;
+import org.gluu.oxd.server.service.PublicOpKeyService;
+import org.jboss.resteasy.client.ClientExecutor;
+import org.jboss.resteasy.client.ClientRequest;
+
+public class OpClientFactoryImpl implements OpClientFactory {
+
+    public OpClientFactoryImpl() {
+    }
+
+    public TokenClient createTokenClient(String url) {
+        return new TokenClient(url);
+    }
+
+    public TokenClient createTokenClientWithUmaProtectionScope(String url) {
+        return new TokenClient(url);
+    }
+
+    public UserInfoClient createUserInfoClient(String url) {
+        return new UserInfoClient(url);
+    }
+
+    public RegisterClient createRegisterClient(String url) {
+        return new RegisterClient(url);
+    }
+
+    public OpenIdConfigurationClient createOpenIdConfigurationClient(String url) throws Exception{
+        return new OpenIdConfigurationClient(url);
+    }
+
+    public AuthorizeClient createAuthorizeClient(String url) {
+        return new AuthorizeClient(url);
+    }
+
+    public ResourceRegistrar createResourceRegistrar(PatProvider patProvider, ServiceProvider serviceProvider) {
+        return new ResourceRegistrar(patProvider, serviceProvider);
```

Comment:
It looks wrong too, we should mock `ServiceProvider` and `PatProvider` provider instead. But if it's more convenient then we can leave it for now.

Reply:
We have left this for now.